domain/infosync/info_test.go

```diff
@@ -20,172 +20,127 @@ import (
-	if err != nil {
-		t.Fatal(err)
-	}
+	info, err := GlobalInfoSyncerInit(ctx, currentID, func() uint64 { return 1 }, client, false)
+	require.NoError(t, err)
 
 	err = info.newTopologySessionAndStoreServerInfo(ctx, owner.NewSessionDefaultRetryCnt)
+	require.NoError(t, err)
 
-	if err != nil {
-		t.Fatal(err)
-	}
+	topology, err := info.getTopologyFromEtcd(ctx)
+	require.NoError(t, err)
+	require.Equal(t, int64(1282967700000), topology.StartTimestamp)
 
-	topo, err := info.getTopologyFromEtcd(ctx)
-	if err != nil {
-		t.Fatal(err)
-	}
-
-	if topo.StartTimestamp != 1282967700000 {
-		t.Fatal("start_timestamp of topology info does not match")
-	}
-	if v, ok := topo.Labels["foo"]; !ok || v != "bar" {
-		t.Fatal("labels of topology info does not match")
-	}
-
-	if !reflect.DeepEqual(*topo, info.getTopologyInfo()) {
-		t.Fatal("the info in etcd is not match with info.")
-	}
+	v, ok := topology.Labels["foo"]
+	require.True(t, ok)
+	require.Equal(t, "bar", v)
+	require.Equal(t, info.getTopologyInfo(), *topology)
 
 	nonTTLKey := fmt.Sprintf("%s/%s:%v/info", TopologyInformationPath, info.info.IP, info.info.Port)
 	ttlKey := fmt.Sprintf("%s/%s:%v/ttl", TopologyInformationPath, info.info.IP, info.info.Port)
 
-	err = util.DeleteKeyFromEtcd(nonTTLKey, cli, owner.NewSessionDefaultRetryCnt, time.Second)
-	if err != nil {
-		t.Fatal(err)
-	}
+	err = util.DeleteKeyFromEtcd(nonTTLKey, client, owner.NewSessionDefaultRetryCnt, time.Second)
+	require.NoError(t, err)
 
 	// Refresh and re-test if the key exists
 	err = info.RestartTopology(ctx)
-	if err != nil {
-		t.Fatal(err)
-	}
+	require.NoError(t, err)
 
-	topo, err = info.getTopologyFromEtcd(ctx)
-	if err != nil {
-		t.Fatal(err)
-	}
+	topology, err = info.getTopologyFromEtcd(ctx)
+	require.NoError(t, err)
 
 	s, err := os.Executable()
-	if err != nil {
-		t.Fatal(err)
-	}
+	require.NoError(t, err)
 
 	dir := path.Dir(s)
+	require.Equal(t, dir, topology.DeployPath)
+	require.Equal(t, int64(1282967700000), topology.StartTimestamp)
+	require.Equal(t, info.getTopologyInfo(), *topology)
 
-	if topo.DeployPath != dir {
-		t.Fatal("DeployPath not match expected path")
-	}
+	// check ttl key
+	ttlExists, err := info.ttlKeyExists(ctx)
+	require.NoError(t, err)
+	require.True(t, ttlExists)
 
-	if topo.StartTimestamp != 1282967700000 {
-		t.Fatal("start_timestamp of topology info does not match")
-	}
+	err = util.DeleteKeyFromEtcd(ttlKey, client, owner.NewSessionDefaultRetryCnt, time.Second)
+	require.NoError(t, err)
 
-	if !reflect.DeepEqual(*topo, info.getTopologyInfo()) {
-		t.Fatal("the info in etcd is not match with info.")
-	}
+	err = info.updateTopologyAliveness(ctx)
+	require.NoError(t, err)
 
-	// check ttl key
-	ttlExists, err := info.ttlKeyExists(ctx)
+	ttlExists, err = info.ttlKeyExists(ctx)
+	require.NoError(t, err)
+	require.True(t, ttlExists)
+}
+
+func (is *InfoSyncer) getTopologyFromEtcd(ctx context.Context) (*topologyInfo, error) {
```

Comment:
What about moving the two methods out of `_test.go`, the package is not with suffix `_test`: there should not be any priv/access problem.

Reply:
Make senses. I'd like to move helper functions to a separated file, but don't find a good name yet. What `_test.go` it should be?

Reply:
I mean moving it into `info.go`. But, yes, TiDB itself is not using that, it is for PD... I did not notice that. Then it is not necessary change request.

Reply:
Fine. I will mark this discussion as resolved then.